be/src/exprs/split_part.cpp

```diff
@@ -170,7 +170,9 @@ StatusOr<ColumnPtr> StringFunctions::split_part(FunctionContext* context, const
             if (split_index(haystack, delimiter, part_number, slice)) {
                 res.append(slice);
             } else {
-                res.append_null();
+                slice.data = haystack.data;
+                slice.size = haystack.size;
+                res.append(slice);
```

Comment:
There are some other invocation of `res.append_null()` in this method, such as L155 and L163. Should check whether need to change them to append an empty.

Reply:
And maybe append `haystack` only when `part_number==1 || -1`? Otherwise, `SELECT SPLIT_PART('hello world', '|', 100)` will also return `hello world`.

Reply:
Hello @ZiheLiu, Many thanks! I have done the modification and expanded test cases. Could you please check again?